### PR TITLE
StringFilterMatchType enum value additions

### PR DESCRIPTION
**Pull request by zcross**
This adds support for the new `NOT_IN` and `IN` variations of `StringFilterMatchType` used within tracing semantic matcher types.

### CODE REVIEW

**Comment by zcross** [comment]
Whoops, this necessitates the addition of another field to the `StringFilter` message type. On it now.

**Review by nhunhu9 — APPROVED** [review]

**Review by aschepis — APPROVED** [review]

File: chronosphere/tfschema/trace_common.go [hunk 1]
```diff
@@ -68,23 +68,20 @@ var TraceSpanFilterListSchema = &schema.Schema{
...
-			},
+			"tag": TraceTagFilterSchema,
 			// This field is still defined for error messaging to existing users, but is deprecated. Prefer tag.
-			"tags": {
-				Type:       schema.TypeList,
-				Optional:   true,
-				Elem:       traceTagFilterSchema,
-				Deprecated: "use tag instead",
-			},
+			"tags":       TraceTagFilterSchema,
```
Comment (prashantv) [inline chronosphere/tfschema/trace_common.go:73]: why this change -- this drops the `Deprecated` field

File: chronosphere/tfschema/trace_common.go [hunk 2]
```diff
@@ -168,7 +165,13 @@ var TraceStringFilterSchema = &schema.Schema{
 			},
 			"value": {
 				Type:     schema.TypeString,
-				Required: true,
+				Optional: true,
+			},
+			"in_values": {
```
Comment (prashantv) [inline chronosphere/tfschema/trace_common.go:170]: nit: it might be a little confusing to see `NOT_IN` and then `IN_VALUES`. should we call it `values` and use OneOf to ensure either value or values is set, and add some custom validation such that values is only set for the multi-value match type?

Comment (prashantv) [reply chronosphere/tfschema/trace_common.go:170]: I see we already use in_values in the public API -- if we think we should use "values" we should do it in both. otherwise, can keep this as-is

**Review by prashantv — COMMENTED** [review]
From a validation perspective, does the server return dry-run errors when the match type doesn't match the values/in-values?

File: chronosphere/tfschema/trace_common.go (hunk 1 above)
Comment (zcross) [reply chronosphere/tfschema/trace_common.go:73]: yeah, I decided the type reuse was cleaner and that the later-stage error message would suffice

I noticed this because on the `main@HEAD` running `make generate` produced a non-empty diff... because I had shed the "shared type" (apparent in intschema's `main` generation function). This had a bit of a code smell in that it produced two separate types and would've required duplicating shared functions for the two nearly-identical input/return types (one corresponding to the variation having been marked deprecated)

File: chronosphere/tfschema/trace_common.go (hunk 2 above)
Comment (zcross) [reply chronosphere/tfschema/trace_common.go:170]: I found `values` to be a little bit to close to the longstanding `value` (single-value) field name; I pondered this and landed on `in_values` to make the connection to the `IN/NOT_IN` match type a bit more explicit.

BTW: regarding "OneOf" I assume you're talking about at the terraform layer (e.g., not public API protos) – does this refer to `ExactlyOneOf` / `AtLeastOneOf`? On that note, why couldn't I have previously used `ExactlyOneOf` to validate usage of consolidated sibling fields in #2 ?

File: chronosphere/tfschema/trace_common.go (hunk 1 above)
Comment (prashantv) [reply chronosphere/tfschema/trace_common.go:73]: Ah I see, this helper will come in handy then:
https://github.com/chronosphereio/terraform-provider-chronosphere/blob/c1157696f91084738bf97456492457c8a5e96192/chronosphere/tfschema/notification_policy.go#L92

this will share the type, but add the deprecated message

File: chronosphere/tfschema/trace_common.go (hunk 2 above)
Comment (prashantv) [reply chronosphere/tfschema/trace_common.go:170]: Yep, I was thinking of `ExactlyOneOf` and `AtLeastOneOf`. They're optional to give the user errors on the client-side, though we are moving towards server-side validation that returns errors as part of dry-run validation. 

If the dry-run validation works correctly, then we probably don't need to encode the same validation in the TF schema.

**Review by prashantv — APPROVED** [review]

File: chronosphere/tfschema/trace_common.go (hunk 2 above)
Comment (zcross) [reply chronosphere/tfschema/trace_common.go:170]: I'm going to triple check the server side validation and will probably at least try adding the exactlyoneof validation for mutual exclusion over `value` and `in_values`

File: chronosphere/tfschema/trace_common.go (hunk 1 above)
Comment (zcross) [reply chronosphere/tfschema/trace_common.go:73]: that looks like exactly what I needed! strangely, it. still seems to yield a new type?

```

+       Tags            []TraceSpanFilterTags `intschema:"tags,optional"`
+}
+
+type TraceSpanFilterTags struct {
+       Key          string              `intschema:"key,optional"`
+       NumericValue *TraceNumericFilter `intschema:"numeric_value,optional,list_encoded_object"`
+       Value        *TraceStringFilter  `intschema:"value,optional,list_encoded_object"`
 }
```

Anyway, this is not actually an issue: the updated business logic only consumes the preferred field and otherwise just checks for the emptiness of the deprecated list element (without inspecting individual elements) so ... 🤷

File: chronosphere/tfschema/trace_common.go (hunk 2 above)
Comment (zcross) [reply chronosphere/tfschema/trace_common.go:172]: the plot thickens:

```
// Only absolute attribute paths, ones starting with top level attribute
	// names, are supported. Attribute paths cannot be accurately declared
	// for TypeList (if MaxItems is greater than 1), TypeMap, or TypeSet
	// attributes. To reference an attribute under a single configuration block
	// (TypeList with Elem of *Resource and MaxItems of 1), the syntax is
	// "parent_block_name.0.child_attribute_name".
```

```
│ Internal validation of the provider failed! This is always a bug
│ with the provider itself, and not a user issue. Please report
│ this bug:
│ 
│ 3 errors occurred:
│ 	* resource chronosphere_trace_tail_sampling_rules: ExactlyOneOf: value
│ references unknown attribute (value) at part (value)
│ 	* resource chronosphere_dataset: ExactlyOneOf: value references unknown
│ attribute (value) at part (value)
│ 	* resource chronosphere_trace_metrics_rule: ExactlyOneOf: value references
│ unknown attribute (value) at part (value)
```

I think I just have to remove this because this nested type is reused everywhere and enumerating the absolute paths in every transitive usage will be error-prone. We'll just rely on server side validation.

**Comment by zcross** [comment]
I've got a passing e2e test internally, merging
